### PR TITLE
Correctly use locks in ErrorMonitor constructor

### LINKED ISSUE #1674: Review of ErrorMonitor constructor in tests
I am not sure why ErrorMonitor constructor locks and unlocks its mutex in its constructor. Does not make sense to me. Who else would use the object if it is not even constructed yet?

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/dbd073615280f0b737e96017c355ca75835abc1e/tests/vkrenderframework.cpp#L29-L34

On the other hand the `ErrorMonitor::Reset()` method does not have locking, while technically a modifying function.

Not sure generally what the locking is useful for. Most (all?) of the tests are single-threaded.

### PR DESCRIPTION
**Pull request by mark-lunarg**
Fixes #1674, @krOoze, is this what you had in mind?

### CODE REVIEW

**Comment by krOoze** [comment]
Yea, though not sure why there is a mutex at all.

Since it does not seem to be used, it is probably buggy elsewhere anyway. E.g. `ErrorMonitor::GetOtherFailureMsgs()` does not get a read lock, but copies out a vector, which might result in garbage.

**Comment by mark-lunarg** [comment]
@krOoze, nobody is very excited about taking out the lock entirely as there are a few mutlithread tests, so I've tried to plug the holes.

**Comment by mark-lunarg** [comment]
@krOoze, objections to this set of changes?

**Comment by krOoze** [comment]
@mark-lunarg I mean... it worked before, so unless you deadlocked yourself it will work still.

Is there any rationale to which method is locked and which isn't? E.g. `VerifyFound()` still does not have locks, but it reads members that are set from methods that do have locks.

**Comment by mark-lunarg** [comment]
@krOoze, eh, only the lame rational that I was just bracketing stuff that was changing state.  If adding it to the others as well will make it more 'correct', I'll do that.

**Comment by krOoze** [comment]
@mark-lunarg If we are aiming for a quick fix, that is probably what should be done. Just need to be careful so one locking method does not call another locking method. E.g. `GetOtherFailureMsgs()` should not be locked, and probably moved to `private`.

File: tests/vkrenderframework.cpp [hunk 1]
```diff
@@ -137,18 +139,28 @@ VkBool32 ErrorMonitor::CheckForDesiredMsg(const char *const msgString) {
 
 vector<string> ErrorMonitor::GetOtherFailureMsgs() const { return other_messages_; }
 
-VkDebugReportFlagsEXT ErrorMonitor::GetMessageFlags() const { return message_flags_; }
+VkDebugReportFlagsEXT ErrorMonitor::GetMessageFlags() {
+    test_platform_thread_lock_mutex(&mutex_);
+    return message_flags_;
+    test_platform_thread_unlock_mutex(&mutex_);
```
Comment (krOoze) [inline tests/vkrenderframework.cpp:145]: You are unlocking past `return`, so you deadlocked yourself.

**Comment by mark-lunarg** [comment]
@krOoze, updated, thanks for all of your help on this.

**Comment by mark-lunarg** [comment]
Failures here are due to a temporarily borked CI setup, passed internal CI.

**Pull request merged**